src/journal.rs

```diff
@@ -225,6 +226,38 @@ impl Journal {
         sd_try!(ffi::sd_journal_get_realtime_usec(self.j, &mut timestamp_us));
         Ok(system_time_from_realtime_usec(timestamp_us))
     }
+
+    /// Adds a match by which to filter the entries of the journal.
+    /// If a match is applied, only entries with this field set will be iterated.
+    pub fn match_add<T: Into<Vec<u8>>>(&self, key: &str, val: T) -> Result<&Journal> {
```

Comment:
I could use an explanation on why `&mut self` isn't needed here: it appears that these filtering functions are mutating internal state of the journal object.

As we're an ffi interface, it's true that we don't need to exactly follow the `&` and `&mut` meanings as provided in rust (I believe C's abi allows us a bit more leeway), but it seems like it'd be useful in this case to make it clear that the journal's internal state is being modified.